packages/ui/src/components/item-layout/item-layout.native.tsx

```diff
@@ -0,0 +1,103 @@
+import { ReactNode, isValidElement } from 'react';
+import { StyleSheet } from 'react-native';
+
+import { Flag } from '../../components/flag/flag.native';
+import { Text } from '../../components/text/text.native';
+import { ChevronUpIcon } from '../../icons/chevron-up-icon.native';
+import { Box } from '../box/box.native';
+
+export interface ItemLayoutProps {
+  captionLeft: ReactNode;
+  captionRight?: ReactNode;
+  flagImg: ReactNode;
+  isDisabled?: boolean;
+  isSelected?: boolean;
+  showChevron?: boolean;
+  titleLeft: ReactNode;
+  titleRight: ReactNode;
+}
+
+export function ItemLayout({
+  captionLeft,
+  captionRight,
+  flagImg,
+  showChevron,
+  titleLeft,
+  titleRight,
+}: ItemLayoutProps) {
+  return (
+    <Flag img={flagImg} spacing="3">
+      <Box style={styles.container}>
+        <Box
```

Comment:
@edgarkhanzadian : I have some Q's here. 

- Do you think its worth making wrappers for native - `Stack` + `HStack`? Then the code would match up better?
- Is `Box` the right element to use here? On web we are aligning `start` and `end` and not `flex-start` and `flex-end`